localstack-core/localstack/services/apigateway/integration.py

```diff
@@ -1118,3 +1126,19 @@ def invoke(self, invocation_context: ApiInvocationContext):
         self.response_templates.render(invocation_context)
         invocation_context.response.headers["Content-Length"] = str(len(response.content or ""))
         return invocation_context.response
+
+
+def apply_integration_request_parameters(
+    uri: str, integration: Dict[str, Any], invocation_context: ApiInvocationContext
+):
+    groups = re.findall(r"({[a-zA-Z]+})", uri)
+    request_parameters = integration.get("requestParameters", {})
+    for group in groups:
+        key = group.replace("{", "").replace("}", "")
+        request_param_key = f"integration.request.path.{key}"
+        associatedValueKey = request_parameters.get(request_param_key)
+        if associatedValueKey:
+            value = invocation_context.context.get(associatedValueKey.replace("context.", ""))
+            if value:
+                uri = uri.replace(group, value)
+    return uri
```

Comment:
Thank you @cyrilhagege for submitting this pr.

I agree with @bentsku, that it would be best to have the feature tested. Also, consider looking into `localstack.utils.json.extract_jsonpath`. This could help with the point raised by Ben.

I also think that the 2 problems at hand here can be solved by using `PROVIDER_OVERRIDE_APIGATEWAY=next_gen`